app/models/job_execution.rb

```diff
@@ -179,26 +197,27 @@ def lock_project(&block)
 
   class << self
     def find_by_id(id)
-      registry[id.to_i]
+      registry.find(id)
     end
 
-    def start_job(reference, job, &block)
-      new(reference, job).tap do |job_execution|
-        if enabled
-          registry[job.id] = job_execution
-          job_execution.start!(&block)
-          ActiveSupport::Notifications.instrument "job.threads", thread_count: registry.length
-        end
-      end
+    def active?(id, key: id)
+      registry.active?(key, id)
+    end
+
+    def queued?(id, key: id)
+      registry.queued?(key, id)
+    end
+
+    def start_job(job_execution, key: job_execution.id)
```

Comment:
this may be better as an instance method rather than a class method that accepts one instance as an argument